src/dotnet/CarbonAware.Aggregators/src/CarbonAware/SciScoreAggregator.cs

```diff
@@ -0,0 +1,53 @@
+using CarbonAware.Model;
+using CarbonAware.Interfaces;
+using Microsoft.Extensions.Logging;
+using System.Collections;
+using System.Globalization;
+
+namespace CarbonAware.Aggregators.SciScore
+{
+    public class SciScoreAggregator : ISciScoreAggregator
+    {
+        private readonly ILogger<SciScoreAggregator> _logger;
+
+        private readonly ICarbonIntensityDataSource _carbonIntensityDataSource;
+
+        public SciScoreAggregator(ILogger<SciScoreAggregator> logger, ICarbonIntensityDataSource carbonIntensityDataSource)
+        {
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this._carbonIntensityDataSource = carbonIntensityDataSource;
+        }
+
+        /// <inheritdoc />
+        public async Task<double> CalculateAverageCarbonIntensityAsync(Location location, string timeInterval)
+        {
+            (DateTimeOffset start, DateTimeOffset end) = this.ParseTimeInterval(timeInterval);
```

Comment:
Actually I just noticed in Akshara's PR that you say this should be and ISO8601 string, so it would be really easy to validate here (or at the start of the other function) and throw immediately if the string doesn't match what we expect

Reply:
Validation is performed by the `ParseTimeInterval` function